src/main/resources/avro/variants.avdl

```diff
@@ -120,30 +168,66 @@ record Call {
   union { null, string } callSetName = null;
 
   /**
-  The genotype of this variant call. Each value represents either the value
-  of the referenceBases field or is a 1-based index into alternateBases.
+  The genotype of this variant call. Must be null if there is an associated
+  `Allele`, which implies a uniform genotype of that `Allele`.
+
+  A 0 value represents the reference allele of the associated `Variant`. Any
+  other value is a 1-based index into the alternate alleles of the associated
+  `Variant`.
+
   If a variant had a referenceBases field of "T", an alternateBases
   value of ["A", "C"], and the genotype was [2, 1], that would mean the call
   represented the heterozygous value "CA" for this variant. If the genotype
   was instead [0, 1] the represented value would be "TA". Ordering of the
   genotype values is important if the phaseset field is present.
+
+  If the API server supports the "classic" mode, and this `Call` is associated
+  with a `Variant`, this field must not be null.
   */
-  array<int> genotype = [];
+  union { null, array<int> } genotype = null;
```

Comment:
It is a bit clunky. We need to be able to not have a `genotype` for a call that's there to give a copy number for an `Allele`. If we split up `Call`s and `AlleleCall`s, this will go away.